modules/apps/object/object-service/src/main/java/com/liferay/object/internal/tree/TreeFactoryImpl.java

```diff
@@ -37,14 +41,33 @@ public class TreeFactoryImpl implements TreeFactory {
 	public Tree createObjectDefinitionTree(long objectDefinitionId)
 		throws PortalException {
 
+		ObjectDefinitionLocalService objectDefinitionLocalService =
+			_objectDefinitionLocalServiceSnapshot.get();
+
 		return _create(
 			objectDefinitionId,
 			node -> TransformUtil.transform(
 				_objectRelationshipLocalService.getObjectRelationships(
 					node.getPrimaryKey(), true),
-				objectRelationship -> new Node(
-					new Edge(objectRelationship.getObjectRelationshipId()),
-					node, objectRelationship.getObjectDefinitionId2())));
+				objectRelationship -> {
+					ObjectDefinition objectDefinition1 =
+						objectDefinitionLocalService.getObjectDefinition(
+							node.getPrimaryKey());
+
+					ObjectDefinition objectDefinition2 =
+						objectDefinitionLocalService.getObjectDefinition(
+							objectRelationship.getObjectDefinitionId2());
+
+					if (objectDefinition1.isApproved() !=
+							objectDefinition2.isApproved()) {
+
+						return null;
+					}
```

Comment:
I'm not confident about this change, let's discuss this together :+1:

Reply:
I believe I can avoid an unnecessary get, we can talk about this later